zerver/views/users.py

```diff
@@ -121,13 +130,26 @@ def deactivate_bot_backend(
     request: HttpRequest, user_profile: UserProfile, bot_id: int
 ) -> HttpResponse:
     target = access_bot_by_id(user_profile, bot_id)
-    return _deactivate_user_profile_backend(request, user_profile, target)
+    return _deactivate_user_profile_backend(request, user_profile, target, None, None, None)
 
 
 def _deactivate_user_profile_backend(
-    request: HttpRequest, user_profile: UserProfile, target: UserProfile
+    request: HttpRequest,
+    user_profile: UserProfile,
+    target: UserProfile,
+    email_text: Optional[str],
+    realm_name: Optional[str],
+    username: Optional[str],
 ) -> HttpResponse:
     do_deactivate_user(target, acting_user=user_profile)
+    if email_text:
+        send_email(
+            "zerver/emails/deactivate",
+            to_emails=[target.email],
+            from_name="Zulip Administrator",
+            from_address=FromAddress.NOREPLY,
+            context={"text": email_text, "realm_name": realm_name, "username": username},
```

Comment:
the realm name is just `user_profile.realm.name`; the username is `user_profile.full_name`.